### ISSUE TITLE
Support newer versions of InfluxDB

### ISSUE THREAD

**Issue by sofixa** [saltstack-formulas/influxdb-formula#23]
Hi, nice formula, but newer versions (0.13 + ) don't work due to the fact that they aren't declared in config_defaults.yaml.

It would be nice to add a fallback with some default parameters so there would be no need to update the formula every time there's a new versio on Influx. What do you think?

Cheers,
Adrian

**Comment by n-rodriguez** [comment]
Hello! I'm working on a new version of this formula. It seems that installing/configuring InlfuxDB has changed a lot since 2016 : https://docs.influxdata.com/influxdb/v1.7/introduction/installation/#installing-influxdb-oss. This new formula is almost the same than the [telegraf-formula]().
Maybe we should add semantic-release and then replace the formula (with all the deprecation things)

**Comment by n-rodriguez** [comment]
@myii now that semantic-release has been merged, how do we do for the deprecation cycle?

**Comment by myii** [comment]
@n-rodriguez Generally the same method as the `php-formula`, in this sort of order (the first two can be merged):

* https://github.com/saltstack-formulas/php-formula/pull/185
* https://github.com/saltstack-formulas/php-formula/pull/187
* https://github.com/saltstack-formulas/php-formula/pull/183

**Comment by n-rodriguez** [comment]
@myii thank you! but I don't know how it could be applied to this formula since almost everything was in `init.sls` or `config.sls`.

**Comment by myii** [comment]
Is #26 the new `v1.0.0`?

**Comment by n-rodriguez** [comment]
> Is #26 the new `v1.0.0`?

yes

**Comment by myii** [comment]
@n-rodriguez Let's have a last version before the upgrading to `v1.0.0`, that shows an upcoming deprecation warning.  Then we can look at #26.

**Comment by WarrenWilkinson** [comment]
It looks ilke the last version of influxdb supported is 0.13 from back in 2016, and that big update pull (#26) was not merged.

So... does this repository still have value?  I'm asking because I'm unsure if  this repo is doing useful work for somebody, or just consuming time for salt maintainers and salt users.

**Comment by myii** [comment]
> It looks ilke the last version of influxdb supported is 0.13 from back in 2016, and that big update pull (#26) was not merged.
> 
> So... does this repository still have value? I'm asking because I'm unsure if this repo is doing useful work for somebody, or just consuming time for salt maintainers and salt users.

@WarrenWilkinson If the effort is available to update and maintain this formula, then I'm sure it can have value.  The general structure is being maintained, since it was converted to `semantic-release`.  The formula specifics need someone to commit to, as shown in the [`CODEOWNERS`](https://github.com/saltstack-formulas/influxdb-formula/blob/0bc456623511153e000f4644f0cffeafc2aa044c/CODEOWNERS) file.

**Comment by myii** [comment]
@n-rodriguez Would you mind if someone builds upon the excellent work you provided in #26?

**Comment by n-rodriguez** [comment]
I'm fine with it

**Comment by myii** [comment]
> I'm fine with it

Thanks, I'll try to get around to it one of these days if no-one else gets there first...

**Comment by myii** [comment]
> 1. a nice, intermediary version with a deprecation warning
> 2. the actual changes, which were done for #26 (and for which the fork has gone see_no_evil )
> 
> Is this right?

@andreimoustache That's right, as I've linked to above in https://github.com/saltstack-formulas/influxdb-formula/issues/23#issuecomment-524638373.  Would you be willing to take this on?  If so, I've got the commits from the fork that's disappeared, I can push those to my fork so that you could get access to them.

**Comment by andreimoustache** [comment]
@myii gotcha! I'll sure try to give it a go at some point next week, but I won't make any promises. If you still have access to the work that's been already done and it's relevant, sure, push them, pity if the work goes to waste! Otherwise, still fine!

**Comment by myii** [comment]
@andreimoustache I found the latest version I had locally and rebased it on the latest version here.  This is the commit:

* https://github.com/myii/influxdb-formula/commit/ea21c93712d8a177b3d98f8ed9085c605d4b6a4b

CC: @n-rodriguez (in case you feel like resurrecting this).

**Comment by andreimoustache** [comment]
@myii ah wow, you found it, awesome! Unfortunately I haven't had time to go into this, hopefully this weekend. Regardless, it's great that you found it! Thank you!

**Comment by myii** [comment]
> @myii ah wow, you found it, awesome! Unfortunately I haven't had time to go into this, hopefully this weekend. Regardless, it's great that you found it! Thank you!

@andreimoustache Don't thank me too much, it's completely failing in Travis in its current form!  Hopefully it isn't too far away from a working solution, though.  Or at least an inspiration towards that.

**Comment by sticky-note** [comment]
@myii @n-rodriguez @andreimoustache What are the news on this ?

**Comment by andreimoustache** [comment]
@sticky-note Unfortunately, that weekend never came, and I've not touched it.
